packages/adapter-components/src/client/pagination.ts

```diff
@@ -158,14 +159,18 @@ export const traverseRequests: (
  */
 export const getWithItemOffsetPagination = ({
   firstIndex,
-  itemsPerPage,
+  queryParamsPageSizeName,
 } : {
   firstIndex: number
-  itemsPerPage: number
+  queryParamsPageSizeName: string | undefined
 }): PaginationFunc => {
   const nextPage: PaginationFunc = ({ page, getParams, currentParams, pageSize }) => {
-    const { paginationField } = getParams
-    if (paginationField === undefined || page.length < pageSize) {
+    const { paginationField, queryParams } = getParams
+
+    const itemsPerPage = queryParamsPageSizeName && queryParams
```

Comment:
queryParams[queryParamsPageSizeName] can be undefined, so we should check something like
```suggestion
    const itemsPerPage = queryParamsPageSizeName !== undefined && queryParams?.[queryParamsPageSizeName] !== undefined
```